lib/keyfile/keyfile.c

```diff
@@ -384,15 +384,15 @@ keyfile_read(const char * filename, uint64_t * machinenum, int keys)
 		warn0("Key file has unreasonable size: %s", filename);
```

Comment:
Rather than adding so many casts, how about adding `size_t keylen`, moving the "unreasonable size" check up a couple lines with a new `/* Validate file size */` comment, and following it by `keylen = (size_t)(sb.st_size);`?